src/components/SelectionList/types.ts

```diff
@@ -108,15 +108,15 @@ type ExtendedTargetedEvent = TargetedEvent & {
     };
 };
 
-type ListItem = {
+type ListItem<K = string> = {
```

Comment:
Let's make this enforce the proper types of react keys, so someone cant use objects, or some other invalid type 

```suggestion
type ListItem<K extends string | number = string> = {
```